radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -46,6 +47,138 @@ pub struct TransactionReceiptV1 {
+        detailed_execution_cost_breakdown: &[(usize, ExecutionCostBreakdownItem)],
+    ) -> String {
+        let network_definition = NetworkDefinition::mainnet();
+        let address_bech32m_encoder = AddressBech32Encoder::new(&network_definition);
+
+        let mut lines = Vec::<String>::new();
+        let mut path_stack = vec![];
+        for (index, (_, execution_item)) in detailed_execution_cost_breakdown.iter().enumerate() {
+            // Constructing the full path
+            match execution_item {
+                ExecutionCostBreakdownItem::Invocation { actor, .. } => {
+                    let actor_string = match actor {
+                        Actor::Root => "root".to_owned(),
+                        Actor::Method(MethodActor {
+                            node_id,
+                            ref ident,
+                            ref object_info,
+                            ..
+                        }) => {
+                            format!(
+                                "Method <{}>::{}::{}",
+                                address_bech32m_encoder
+                                    .encode(node_id.as_bytes())
+                                    .expect("Encoding of an address can't fail"),
+                                object_info.blueprint_info.blueprint_id.blueprint_name,
+                                ident
+                            )
+                        }
+                        Actor::Function(FunctionActor {
+                            ref blueprint_id,
+                            ref ident,
+                            ..
+                        }) => {
+                            format!(
+                                "Function <{}>::{}::{}",
+                                address_bech32m_encoder
+                                    .encode(blueprint_id.package_address.as_bytes())
+                                    .expect("Encoding of an address can't fail"),
+                                blueprint_id.blueprint_name,
+                                ident
+                            )
+                        }
+                        Actor::BlueprintHook(BlueprintHookActor {
+                            hook,
+                            ref blueprint_id,
+                            ..
+                        }) => {
+                            format!(
+                                "Blueprint Hook <{}>::{}::{:?}",
+                                address_bech32m_encoder
+                                    .encode(blueprint_id.package_address.as_bytes())
+                                    .expect("Encoding of an address can't fail"),
+                                blueprint_id.blueprint_name,
+                                hook
+                            )
+                        }
+                    };
+                    path_stack.push(format!("Invocation: {actor_string} ({index})"))
```

Comment:
> The names are rather long/noisy - I don't think we need Invocation or index here.
> I guess adding the index prevents the flamegraph merging common parents and makes the flamegraph into a timeline - BUT because it's not a PREFIX, it doesn't give the right ordering. So it's like the worst of both worlds.

The main reason why we have the index in there is to use it if further analysis is needed. As an example, we notice that some "read substate" is consuming too much cost units so we use the `index` in the flamegraph to identify which one it is in the `detailed_execution_cost_breakdown` and analyze it further. 

As you mentioned, in a flamegraph the order is not preserved so doing further analysis without having the index is going to be rather difficult. 

> Perhaps we can support a few modes:

I do not disagree, but would recommend first seeing what modes we _need_ during our analysis of the execution cost units and add them on need basis :)